### ISSUE TITLE
podman exec into krun'd container is not in entering VM

### ISSUE THREAD

**Issue by adrecord** [containers/podman#16843]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

`exec`ing into an existing container using the krun oci runtime is not executing within the microvm.

**Steps to reproduce the issue:**

1. In one terminal, show the current host kernel, then start a container with the krun oci runtime, and show that the kernel inside the container is indeed different, as it's running in the  microvm.

```
user@host-machine$ uname -a
Linux host-machine 6.0.0-5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.10-2 (2022-12-01) x86_64 GNU/Linux
user@host-machine$ podman --runtime krun run --rm -it --name testctr alpine /bin/sh
/ # uname -a
Linux localhost 6.0.6 #1 SMP PREEMPT_DYNAMIC Wed Nov 2 04:44:41 PM UTC 2022 x86_64 Linux
```

2. With the container from the previous step still running, open another terminal and `podman exec` into the container to see what it's kernel reports.

```
user@host-machine$ podman exec -it testctr /bin/sh
/ # uname -a
Linux cf1c2b51a8ac 6.0.0-5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.10-2 (2022-12-01) x86_64 Linux
/ # exit
```

**Describe the results you received:**

The result I observe is that `podman exec` is using the host's kernel, and not that of the microvm that the container was launched with.  Note that running `podman --runtime krun exec -it testctr /bin/sh` exhibits the same behavior.

**Describe the results you expected:**

I expected that `podman exec` in step 2 would be running in the microvm created by krun during the initial `podman run`, and so would report the kernel version of the microvm instead of the host's kernel.

**Additional information you deem important (e.g. issue happens only occasionally):**

This happens all the time.

**Output of `podman version`:**

```
$ podman version
Client:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.18.8
Built:        Wed Dec 31 19:00:00 1969
OS/Arch:      linux/amd64
```

**Output of `podman info`:**

```
$ podman info
host:
  arch: amd64
  buildahVersion: 1.28.0
  cgroupControllers:
  - memory
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: Unknown
    path: /usr/local/lib/podman/conmon
    version: 'conmon version 2.1.5, commit: c9f7f19eb82d5b8151fc3ba7fbbccf03fdcd0325'
  cpuUtilization:
    idlePercent: 98.72
    systemPercent: 0.48
    userPercent: 0.8
  cpus: 8
  distribution:
    codename: bookworm
    distribution: debian
    version: unknown
  eventLogger: file
  hostname: beast3
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1002
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.0.0-5-amd64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 17514094592
  memTotal: 33635688448
  networkBackend: cni
  ociRuntime:
    name: crun
    package: Unknown
    path: /usr/local/bin/crun
    version: |-
      crun version 1.6
      commit: 18cf2efbb8feb2b2f20e316520e0fd0b6c41ef4d
      spec: 1.0.0
      +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +LIBKRUN +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_MKNOD,CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: ""
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/local/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.2
  swapFree: 34324082688
  swapTotal: 34324082688
  uptime: 101h 8m 50.00s (Approximately 4.21 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /data/home/andy/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 1
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.ignore_chown_errors: "true"
    overlay.mount_program:
      Executable: /usr/local/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fuse-overlayfs: version 1.9
        fusermount3 version: 3.12.0
        FUSE library version 3.12.0
        using FUSE kernel interface version 7.31
    overlay.mountopt: nodev,fsync=0
  graphRoot: /data/home/andy/.local/share/containers/storage
  graphRootAllocated: 4000787030016
  graphRootUsed: 1147891400704
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 7
  runRoot: /run/user/1000/containers
  volumePath: /data/home/andy/.local/share/containers/storage/volumes
version:
  APIVersion: 4.3.1
  Built: 0
  BuiltTime: Wed Dec 31 19:00:00 1969
  GitCommit: ""
  GoVersion: go1.18.8
  Os: linux
  OsArch: linux/amd64
  Version: 4.3.1
```

**Have you tested with the latest version of Podman and have you checked [the Podman Troubleshooting Guide](https://github.com/containers/podman/blob/main/troubleshooting.md)?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

n/a

**Comment by mheon** [comment]
Seems likely to be a crun issue? @giuseppe

**Comment by giuseppe** [comment]
krun doesn't support exec'ing into the container, I think we can close this issue.

@slp FYI

**Comment by adrecord** [comment]
Shouldn't the `podman exec` fail if krun doesn't support exec?  The current behavior gives the user the impression that it works, causing the user to start processes in the exec that they mistakenly believe to be sandboxed in the microvm.  It seems like a security issue to allow this exec under this false assumption.

**Comment by giuseppe** [comment]
could there be cases where it is helpful to be in the krun "container" instead of the VM?  If we error out, then we miss the possibility to be in that environment

**Comment by adrecord** [comment]
I personally don't see a benefit of that.  It just seems confusing as all docs for `exec`ing into a running container describe it as starting a new process in the same namespaces as the already running container, which is not what's happening here.

The way I discovered this was by execing into a krun container and starting other processes.  I then saw these processes show up in my host's process list, which was unexpected, and actually kinda scary.

That said, overall I'm loving the libkrun integration and having good luck with it, so thanks very much for all the great work on it!  As far as I know krun is now the only OCI runtime with vm isolation since kata sadly removed OCI support.
